test/e2e/framework/addon/chart/addon.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/jetstack/cert-manager/test/e2e/framework/config"
+)
+
+// Chart is a generic Helm chart addon for the test environment
+type Chart struct {
+	config        *config.Config
+	tillerDetails *tiller.Details
+	// temporary directory used as the --home flag to Helm
+	home string
+
+	// Tiller is the tiller instance to submit the release to
+	Tiller *tiller.Tiller
+
+	// ReleaseName for this Helm release
+	// `helm install --name {{ReleaseName}}`
+	ReleaseName string
+
+	// Namespace for the Helm release
+	// `helm install --namespace {{Namespace}}`
+	Namespace string
+
+	// ChartName is the name of the chart to deploy
+	// `helm install {{ChartName}}``
+	ChartName string
+
+	// ChartVersion is the version of the chart to deploy
+	// `helm install --version {{ChartVersion}}`
+	ChartVersion string
+
+	// Vars are additional --set arguments for helm install
+	// `helm install --set {{Vars[0].Key}}={{Vars[0].Value}} --set {{Vars[1].Key}}={{Vars[1].Value}} ...`
+	Vars []StringTuple
+
+	// Values is a list of paths to additional values.yaml files to include
+	// `helm install --values {{Values[0]}} --values {{Values[1]}} ...`
+	Values []string
+
+	// If UpdateDeps is true, 'helm dep update' will be run against the chart
+	// before installing.
+	// This should only be set to true when the ChartName is a local path on disk.
+	UpdateDeps bool
+}
+
+// StringTuple is a tuple of strings, used to create ordered maps
+type StringTuple struct {
+	Key   string
+	Value string
+}
+
+// Details return the details about the Tiller instance deployed
+type Details struct {
+	// Helm chart release name
+	ReleaseName string
+
+	// Namespace that Tiller has been deployed into
+	Namespace string
+}
+
+func (c *Chart) Setup(cfg *config.Config) error {
+	var err error
+
+	if c.Tiller == nil {
+		return fmt.Errorf("tiller base addon must be provided")
+	}
+	c.tillerDetails, err = c.Tiller.Details()
+	if err != nil {
+		return err
+	}
+
+	c.home, err = ioutil.TempDir("", "helm-chart-install")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Provision an instance of tiller-deploy
+func (c *Chart) Provision() error {
+	err := c.runHelmClientInit()
+	if err != nil {
+		return fmt.Errorf("error running 'helm init': %v", err)
+	}
+
+	if c.UpdateDeps {
+		err := c.runDepUpdate()
+		if err != nil {
+			return fmt.Errorf("error updating helm chart dependencies: %v", err)
+		}
+	}
+
+	err = c.runInstall()
+	if err != nil {
+		return fmt.Errorf("error install helm chart: %v", err)
+	}
+
+	return nil
+}
+
+func (c *Chart) runHelmClientInit() error {
+	err := c.buildHelmCmd("init", "--client-only").Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Chart) runDepUpdate() error {
+	err := c.buildHelmCmd("dep", "update", c.ChartName).Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Chart) runInstall() error {
+	args := []string{"install", c.ChartName,
+		// "--wait",
```

Comment:
Had some issues with helm not ever exiting when using `--wait` so I've disabled it for now